src/osgWrappers/serializers/osg/TextureBuffer.cpp

```diff
@@ -0,0 +1,98 @@
+#include <osg/TextureBuffer>
+#include <osg/Array>
+#include <osgDB/ObjectWrapper>
+#include <osgDB/InputStream>
+#include <osgDB/OutputStream>
+
+static bool checkImage( const osg::TextureBuffer& texture )
+{
+    return texture.getImage() != NULL;
+}
+
+static bool readImage( osgDB::InputStream& is, osg::TextureBuffer& texture )
+{
+    if ( is.isBinary() )
+    {
+        osg::ref_ptr<osg::Image> image = is.readImage();
+        texture.setImage(image.get());
+    }
+    else
+    {        
+        is >> is.BEGIN_BRACKET;
+        osg::ref_ptr<osg::Image> image = is.readImage();
+        texture.setImage(image.get());
+        is >> is.END_BRACKET;    
+    }    
+    
+    return true;
+}
+
+static bool writeImage( osgDB::OutputStream& os, const osg::TextureBuffer& texture )
+{
+    const osg::Image* image = texture.getImage();    
+
+    if ( os.isBinary() )
+    {        
+        os.writeImage(image);
+    }
+    else
+    {
+        os << os.BEGIN_BRACKET << std::endl;
+        os.writeImage(image);
+        os << os.END_BRACKET << std::endl;
+    }
+
+    return true;
+}
+
+
+static bool checkBufferData( const osg::TextureBuffer& texture )
+{
+    return texture.getBufferData() != NULL && texture.getBufferData()->asArray() != NULL;
+}
+
+static bool readBufferData( osgDB::InputStream& is, osg::TextureBuffer& texture )
+{
+    if ( is.isBinary() )
+    {
+        osg::ref_ptr<osg::Array> bufferData = is.readArray();
+        texture.setBufferData(bufferData);
+    }
+    else
+    {
+        is >> is.BEGIN_BRACKET;
+        osg::ref_ptr<osg::Array> bufferData = is.readArray();
+        texture.setBufferData(bufferData);
+        is >> is.END_BRACKET;    
+    }
+    
+    return true;
+}
+
+static bool writeBufferData( osgDB::OutputStream& os, const osg::TextureBuffer& texture )
+{    
+    const osg::Array* array = texture.getBufferData()->asArray();
+
+    if ( os.isBinary() )
+    {
+        os.writeArray(array);
+    }
+    else
+    {        
+        os << os.BEGIN_BRACKET << std::endl;
+        os.writeArray(array);
+        os << os.END_BRACKET << std::endl;
+    }    
+
+    return true;
+}
+
+REGISTER_OBJECT_WRAPPER( TextureBuffer,
+                         new osg::TextureBuffer,
+                         osg::TextureBuffer,
+                         "osg::Object osg::StateAttribute osg::Texture osg::TextureBuffer" )
+{
+    ADD_USER_SERIALIZER(Image);            // _bufferData as osg::Image
```

Comment:
I looked into using the _osg::InputStream::readObject_/_OutputStream::writeObject_ methods:
It seems to work fine for Array data but there is an issue if _osg::Image_ is used. The serializer for _osg::Image_ does not store the image data itself if storing data inline is enabled. The logic for this is implemented in the _OutputStream::writeImage_ method and not in the serializer itself. Storing image data inline comes in handy if special image formats are used(float textures, integer textures).
So it would be necessary to adapt the serializer for osg::Image as well.